lib/core/plugins/manager.js

```diff
@@ -709,25 +716,63 @@ class PluginsManager {
     }
   }
 
-  /**
-   * @param {object} plugin
-   * @param {number} pipeWarnTime
-   * @param {number} pipeTimeout
-   */
-  _initPipes (plugin, pipeWarnTime, pipeTimeout) {
-    const methodsList = getMethods(plugin.object);
+  async beforeDocumentWrite (pluginName, index, collection, filters, callback) {
+    debug('[%s] Register beforeDocumentWrite callback for "%s:%s" with %o', pluginName, index, collection, filters);
 
-    let
-      _warnTime = pipeWarnTime,
-      _timeout = pipeTimeout;
+    const count = await this._documentPipes.registerOnWrite(
+      pluginName,
+      index,
+      collection,
+      filters,
+      callback);
 
-    if (plugin.config && plugin.config.pipeWarnTime !== undefined) {
-      _warnTime = plugin.config.pipeWarnTime;
+    if (count === 1) {
```

Comment:
If a similar conditional pipe is registered by another plugin (e.g. a condition on the index name), then the room id returned by Koncorde will be identical for that plugin, and the count returned will be > 1.

Reply:
Good catch :+1: